### PR TITLE
Add support for running with Docker

### PR DESCRIPTION
**Pull request by wesleylancel**
Credits to @bshyong, borrowed most of this from his work on Really Good Emails.

### CODE REVIEW

**Comment by kenrick** [comment]
:+1:

File: config/database.yml.sample [hunk 1]
```diff
@@ -29,15 +29,15 @@ development:
...
 
   # The password associated with the postgres role (username).
-  #password:
+  password: postgres
 
   # Connect on a TCP socket. Omitted by default since the client uses a
   # domain socket that doesn't need configuration. Windows does not have
   # domain sockets, so uncomment these lines.
-  #host: localhost
+  host: db
```
Comment (vanstee) [inline config/database.yml.sample]: Wonder if there's a good way to make this appropriate for normal local dev as well?

File: README.md [hunk 2]
```diff
@@ -15,3 +15,28 @@ Visit [https://assembly.com](https://assembly.com) to learn more.
...
+- Build the Docker containers
+  - $ `fig build`
+- Run the app
+  - $ `fig up`
+- Migrate the database
+  - Do this in another terminal window
+    - `fig run web rake db:create`
+- Access the app locally
+  - Run `boot2docker ip` to get the ip address of the Docker daemon
+  - View the app at `IP_ADDRESS:3000`
```
Comment (vanstee) [inline README.md]: I bet a script like our `bin/setup` script would be super nice here.

File: config/database.yml.sample (hunk 1 above)
Comment (wesleylancel) [inline config/database.yml.sample]: @vanstee Only thing I can think of is including an instruction in the README for either changing to `db` or `localhost`.

Comment (vanstee) [inline config/database.yml.sample]: We could do this heroku style and just ignore the database.yml by including the DATABASE_URL env var when running the server.

Comment (vanstee) [inline config/database.yml.sample]: I'll merge this and figure that out later.

File: README.md (hunk 2 above)
Comment (wesleylancel) [inline README.md]: Yeah, might be nice, though a fair amount of people should already have boot2docker installed and maybe even fig.

**Comment by wesleylancel** [comment]
Thanks for the merge.

**Pull request merged**